src/deepqmc/log.py

```diff
@@ -200,14 +202,15 @@ class H5Logger:
     def __init__(
         self,
         workdir: str,
+        observable_names: Optional[list[str]] = None,
         *,
         keys_whitelist: Optional[list[str]] = None,
         init_step: int = 0,
         aux_data: Optional[dict] = None,
     ):
         self.keys_whitelist = (
             keys_whitelist if keys_whitelist is not None else ['local_energy']
-        )
+        ) + (observable_names or [])
```

Comment:
Why do we need a new argument here? Couldn't the `observable_names` be merged with the `keys_whitelist` before passing it to the `H5Logger`? As far as I see they are mostly equivalent. Also `'local_energy'` should always be in `observable_names` and the logic could potentially be further simplified here.

Reply:
The problem is that the variable `keys_whitelist` doesn't exist in `train.py`, because we pass with `h5_logger_constructor = partial(H5Logger, keys_whitelist=[...])`. So we'd have to tinker with `h5_logger_constructor.kwargs` but that's pretty messy.

Reply:
As discussed, renamed `observable_names` to `additional_keys_to_whitelist` in https://github.com/deepqmc/deepqmc/pull/216/commits/55e13abc6224c2883a9fae091068b0f0a46a38aa